components/brave_wallet/browser/swap_service.cc

```diff
@@ -216,36 +215,13 @@ GURL AppendJupiterQuoteParams(const GURL& swap_url,
   return url;
 }
 
-base::flat_map<std::string, std::string> GetHeaders() {
-  return {{kBraveServicesKeyHeader, BUILDFLAG(BRAVE_SERVICES_KEY)}};
-}
-
-std::string GetBaseSwapURL(const std::string& chain_id) {
-  if (chain_id == brave_wallet::mojom::kSepoliaChainId) {
-    return brave_wallet::kZeroExSepoliaBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kMainnetChainId) {
-    return brave_wallet::kZeroExEthereumBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kPolygonMainnetChainId) {
-    return brave_wallet::kZeroExPolygonBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kBnbSmartChainMainnetChainId) {
-    return brave_wallet::kZeroExBinanceSmartChainBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kAvalancheMainnetChainId) {
-    return brave_wallet::kZeroExAvalancheBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kFantomMainnetChainId) {
-    return brave_wallet::kZeroExFantomBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kCeloMainnetChainId) {
-    return brave_wallet::kZeroExCeloBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kOptimismMainnetChainId) {
-    return brave_wallet::kZeroExOptimismBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kArbitrumMainnetChainId) {
-    return brave_wallet::kZeroExArbitrumBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kSolanaMainnet) {
-    return brave_wallet::kJupiterBaseAPIURL;
-  } else if (chain_id == brave_wallet::mojom::kBaseMainnetChainId) {
-    return brave_wallet::kZeroExBaseBaseAPIURL;
+base::flat_map<std::string, std::string> GetHeaders(bool is_zero_ex = false) {
```

Comment:
Let's not use a default value. Either no default value or separate GetHeadersForZeroEx would be better

Reply:
[review(supermassive): separate function for 0x headers](https://github.com/brave/brave-core/pull/26231/commits/21ac0e1a62e536a5275548b08222ba1ca01ecd78)